src/app/views.py

```diff
@@ -857,58 +851,46 @@ def license_xml_edit(request, page_id):
     else:
         return HttpResponseRedirect('/app/xml_upload')
 
+
+def get_context_dict(request, license_obj):
+    context_dict = {}
+    if request.user.is_authenticated:
+        user = request.user
+        try:
+            github_login = user.social_auth.get(provider="github")
+        except UserSocialAuth.DoesNotExist:
+            github_login = None
+        context_dict["github_login"] = github_login
+    context_dict["xml_text"] = license_obj.xml
+    context_dict["license_name"] = license_obj.fullname
+    return context_dict
+
+
 def edit_license_xml(request, license_id=None):
     """View for editing the XML file corresponsing to a license entry
-    returns editor.html """
-    context_dict = {}
-    ajaxdict = {}
+    returns editor.html"""
     if license_id:
         if not LicenseRequest.objects.filter(id=license_id).exists():
-            return render(request,
-                '404.html',context_dict,status=404
-                )
-        if request.user.is_authenticated:
-            user = request.user
-            try:
-                github_login = user.social_auth.get(provider='github')
-            except UserSocialAuth.DoesNotExist:
-                github_login = None
-            context_dict["github_login"] = github_login
+            return render(request, "404.html", {}, status=404)
         license_obj = LicenseRequest.objects.get(id=license_id)
-        context_dict["xml_text"] = license_obj.xml
-        context_dict["license_name"] = license_obj.fullname
-        return render(request,
-            'app/editor.html',context_dict,status=200
-            )
+        context_dict = get_context_dict(request, license_obj)
+        return render(request, "app/editor.html", context_dict, status=200)
     else:
-        return HttpResponseRedirect('/app/license_requests')
+        return HttpResponseRedirect("/app/license_requests")
```

Comment:
why this change is required?

Reply:
It is due to the formatting I have done to the code. If single quotes are more preferred than double quotes I can revert that change.

Reply:
Let's keep the same formatting for now and minimal changes.

There's a different issue tracking for adding linters to the repository which should remove these issues.

Reply:
Updated with [d53ed24](https://github.com/spdx/spdx-online-tools/pull/441/commits/d53ed2496ae73af90a6e26f6ccef74378e2511de).